### PR TITLE
Rdbms destination v2

### PR DESCRIPTION
**Pull request by T-Dnzt**
Create a new `Rdbms` destination to handle delete before inserting. The previous `Rdbms` destination was renamed to `RdbmsWriter` since it only inserts records. Merged Refactor-bq-src to avoid Rubocop spam for unrelated code.

This destination will be used for Farm Cubes (to begin with) to delete all the records matching a specific set of conditions in the Postgresql database used by Tableau. This destination will then insert the records just like the Rdbms Writer. With this solution, Tableau only have the latest data to display.

### CODE REVIEW

File: lib/optimus_prime/destinations/rdbms.rb [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
-require 'pry-byebug'
-
 module OptimusPrime
   module Destinations
-    class Rdbms < Destination
-      def initialize(dsn:, table:, **options)
-        @db = Sequel.connect(dsn, **options)
-        @table = @db[table.to_sym]
+    class Rdbms < RdbmsWriter
+      def initialize(dsn:, table:, conditions:, **options)
```
Comment (Anumodana) [inline lib/optimus_prime/destinations/rdbms.rb]: In my opinion, you should rename `conditions` to make it more specific and states that it's the condition for deleting data.

**Comment by omarkhan** [comment]
Looks good. Can you add a short comment to explain why we need this and the expected use case?

**Comment by T-Dnzt** [comment]
I just added the use case in the PR description. Not sure it's clear though...

File: lib/optimus_prime/destinations/rdbms.rb (hunk 1 above)
Comment (T-Dnzt) [inline lib/optimus_prime/destinations/rdbms.rb]: Alright, I'll change that!

File: spec/optimus_prime/destinations/rdbms_spec.rb [hunk 2]
```diff
@@ -25,26 +32,51 @@
...
+    table.all
   end
 
-  it 'should upload insert records into database' do
-    insert_records
-    expect(records_from_db).to eq(input)
+  context 'string condition' do
+    it 'should delete all the records with ios platform' do
+      insert_records
+      expect(table.where(platform: 'ios').count).to eq 1
```
Comment (Anumodana) [inline spec/optimus_prime/destinations/rdbms_spec.rb]: Have you already check whether `platform` and `Platform` refer to the same column?

Comment (T-Dnzt) [inline spec/optimus_prime/destinations/rdbms_spec.rb]: As we talked about, it depends on the RDBMS and how it's configured. I fixed the capitalized typo to keep everything coherent.

**Comment by Anumodana** [comment]
Other than my comment, it looks good. :+1:

**Pull request merged**